receiver/telemetrygeneratorreceiver/metrics_generator.go

```diff
@@ -18,22 +18,34 @@ import (
 	"math"
 	"math/rand"
 
+	"github.com/mitchellh/mapstructure"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.uber.org/zap"
 )
 
-// hostMetricsGenerator is a generator for host metrics. It generates a sampling of host metrics
-// emulating the Host Metrics receiver: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver
+// metricsGenerator is a generator for metrics. It generates a sampling of metrics
+// based on the configuration provided, adjusting the values & times of the metrics
+// with each generation.
 type metricsGenerator struct {
 	otlpGenerator
 	// dataPointUpdaters is a list of functions that update the data points with new values
 	// for each batch of metrics that are generated
 	dataPointUpdaters []func()
 }
 
+// convenience struct for unmarshalling the metrics config
```

Comment:
nit: Go comments for a struct/interface/function/method need to start with the name.